### ISSUE TITLE
(Suggestion) Disable already booked time slots

### ISSUE THREAD

**Issue by flyingcodeman** [TylerAHolden/react-schedule-meeting#62]
Is there a way to remove/disable already booked time slots within the prop availableTimeslots? If no, this would be a nice and handy feature, especially in combination with a backend to retrieve data/datetime for already booked slots 🙂

**Comment by TylerAHolden** [comment]
When an event is selected a [StartTimeEventEmit](https://react-schedule-meeting.netlify.app/docs/types/#starttimeeventemit) object is passed which includes the available time slots split with the event selected removed so you can just concatenate and save that in your state to pass back in to the component. Therefore the new availableTimeslots prop is without the event selected.

**Comment by flyingcodeman** [comment]
My thoughts are as follows: i have an API, which provides me with time slots that have already been booked (duration: 30 minutes with _eventStartTimeSpreadInMinutes = 0_ ) before - for example:
```
{
	"datetime": "2023-01-12 07:30:00“,
	"datetime": "2023-01-12 18:00:00“,
	"datetime": "2023-01-14 11:00:00“,
	"datetime": "2023-01-14 16:30:00“,
	"datetime": "2023-01-15 09:00:00“
}
```
I would like to hand these over to the meeting component before loading and hide/disable them for the user in advance. Is this already possible within the _StartTimeEventEmit_ object? If yes, how? 🧐

**Comment by TylerAHolden** [comment]
So if I understand, you are suggesting something like an excludeTimeslots prop? Currently, no that is not possible there is only an availableTimeslots prop which is used to derive the events. You would want to be able to pass in an array of available timeslots AND an array of timeslots that are unavailable? - Then you would expect the meeting component to derive the "real" available timeslots?

**Comment by flyingcodeman** [comment]
Exactly. Using the current implementation and related props, I would suggest keeping the current definition of _startTime_ and _endTime_ for the available range of slots AND add an array for unavailable slots - as you said, to only show the **real** and available times lots in the meeting component ☺️

**Comment by flyingcodeman** [comment]
The implementation might also be interesting in two ways: on the one hand, hiding explicit time slots as mentioned (e.g. from an API), but on the other hand also certain time ranges (e.g. a lunch break between 12:00 and 14:00)

**Comment by flyingcodeman** [comment]
Sounds pretty logical! A helper function this way would be a really nice feature - at least for me 🫣
Btw. thanks for your quick and kind replies!

**Comment by flyingcodeman** [comment]
Thank you very much for this quick feature integration & release! 

While importing as described in your [docs](https://master--react-schedule-meeting.netlify.app/docs/utility-functions) _import { timeslotDifference } from 'react-schedule-meeting'_ I am getting an build error with: 
_Attempted import error: 'timeslotDifference' is not exported from 'react-schedule-meeting' (imported as 'timeslotDifference')._
I already deleted _node_modules_ and _package.lock.json_, reinstalled all packages and rebuilt my app - what I am doing wrong?

**Comment by TylerAHolden** [comment]
oh its typo! The docs refer to the function as `timeslotDifference` but the function is exported as `timeSlotDifference` - note the capital S. I will fix the docs for that but all you need to do is change the name
